bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookies/CorrectEnsemblePlacementCommand.java

```diff
@@ -0,0 +1,321 @@
+                            IOUtils.confirmPrompt("Are you sure to relocate target ledgers?");
+                    if (!confirm) {
+                        LOG.error("Relocation is aborted.");
+                        return false;
+                    }
+                }
+            } catch (IOException e) {
+                LOG.error("Error during relocation", e);
+                return false;
+            }
+
+            final ClientConfiguration clientConf = new ClientConfiguration(conf);
+            final BookKeeper bookKeeper = new BookKeeper(clientConf);
+            final BookKeeperAdmin admin = new BookKeeperAdmin(clientConf);
+            return relocate(conf, flags, bookKeeper, admin);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    @VisibleForTesting
+    public boolean relocate(ServerConfiguration conf, CorrectEnsemblePlacementFlags flags,
+                            BookKeeper bookKeeper, BookKeeperAdmin admin) throws Exception {
+        final EnsemblePlacementPolicy placementPolicy = bookKeeper.getPlacementPolicy();
+
+        @Cleanup
+        final MetadataBookieDriver metadataDriver = instantiateMetadataDriver(conf);
+        @Cleanup
+        final LedgerManagerFactory lmf = metadataDriver.getLedgerManagerFactory();
+        @Cleanup
+        final LedgerUnderreplicationManager lum = lmf.newLedgerUnderreplicationManager();
+
+        final List<Long> targetLedgers =
+                flags.ledgerIds.stream().parallel().distinct().filter(ledgerId -> {
+                    try {
+                        return (!flags.skipOpenLedgers || bookKeeper.isClosed(ledgerId))
+                                && !lum.isLedgerBeingReplicated(ledgerId);
+                    } catch (BKException | InterruptedException | ReplicationException e) {
+                        LOG.warn("Failed to add the ledger {} to target.", ledgerId, e);
+                        return false;
+                    }
+                }).collect(Collectors.toList());
+
+        if (targetLedgers.isEmpty()) {
+            LOG.info("None of ledgers are relocated.");
+            return true;
+        }
+
+        final NavigableSet<Long> unAcquirableLedgers = new TreeSet<>();
+        final NavigableSet<Pair<Long, Long>> failedTargets = new ConcurrentSkipListSet<>();
+        final CountDownLatch latch = new CountDownLatch(targetLedgers.size());
+        for (long ledgerId : targetLedgers) {
+            if (!flags.dryRun) {
+                try {
+                    lum.acquireUnderreplicatedLedger(ledgerId);
+                } catch (ReplicationException e) {
+                    LOG.warn("Failed to acquire ledger to under replicated {}.", ledgerId);
+                    unAcquirableLedgers.add(ledgerId);
+                    latch.countDown();
+                    continue;
+                }
+            }
+            admin.asyncOpenLedger(ledgerId, (rc, lh, ctx) -> {
```

Comment:
We should move all this block into some BookKeeperAdmin method.
It will be easier to write tests and also to use this feature programmatically